e2e/cli-smoke-test.sh

```
@@ -436,6 +454,35 @@ check_eq "symlink appears in ls /" "$symlink_present" "true"
+  if [[ "$hardlink_present" == "true" ]]; then
+    break
+  fi
+  sleep "$CLI_RETRY_SLEEP_S"
+done
+check_eq "hardlink appears in ls /" "$hardlink_present" "true"
+
+hardlink_body="$(drive9_retry_read fs cat "$SMALL_HARDLINK")"
+check_eq "cat hardlink returns source content" "$hardlink_body" "cli-smoke-${TS}"
+
+hardlink_nlink="$(cli_stat_field "$SMALL_HARDLINK" "nlink")"
+check_cmd "hardlink stat reports nlink >= 2" bash -c 'test "${1:-0}" -ge 2' -- "$hardlink_nlink"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)